### PR TITLE
Specs: Add the ability to upload the accreditation excel file #13863

### PR DESCRIPTION
**Pull request by zurbergram**
## Description

Unit tests and cleanup after https://github.com/department-of-veterans-affairs/gibct-data-service/pull/751

Pulled out requirements message generation to a separate class to reduce duplicate code

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by ericboehs — APPROVED** [review]

File: app/utilities/upload_types/upload_requirements.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    def validation_messages_inclusion(type)
+      inclusion = []
+
+      type.validators.map do |validations|
+        next unless validations.class == ActiveModel::Validations::InclusionValidator
+
+        array = { message: affected_attributes(validations, type).join(', '),
+                  value: inclusion_requirement_message(validations) }
+        inclusion.push(array)
```
[the remark this answers is not in the document]
Comment (zurbergram) [reply app/utilities/upload_types/upload_requirements.rb]: Cleaned up the code

File: app/utilities/upload_types/upload_requirements.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    end
+
+    def validation_messages_inclusion(type)
+      inclusion = type.validators.map do |validations|
+        next unless validations.class == ActiveModel::Validations::InclusionValidator
+
+        { message: affected_attributes(validations, type).join(', '),
+          value: inclusion_requirement_message(validations) }
+      end
+      inclusion unless inclusion.empty?
```
Comment (ericboehs) [inline app/utilities/upload_types/upload_requirements.rb:19]:
```suggestion
      type.validators.map do |validations|
        next unless validations.class == ActiveModel::Validations::InclusionValidator

        { message: affected_attributes(validations, type).join(', '),
          value: inclusion_requirement_message(validations) }
      end.presence
```

You could also use `presence` to check emptiness.

Comment (zurbergram) [reply app/utilities/upload_types/upload_requirements.rb]: Switched to using `presence`, map wasn't working correct so switched to each

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Pull request merged**